### PR TITLE
[FIX] mail: use correct base url

### PR DESCRIPTION
**Pull request by reth-odoo**
When rendering a template in the render mixin
urls are converted from local to full without consideration for the website of the record being processed.

We now pass the base url of the record, fetched with `get_base_url` so that email links link to the proper website when needed.

task-4104753

### CODE REVIEW

File: addons/mail/tests/test_mail_render.py [hunk 1]
```diff
@@ -211,6 +212,23 @@ def test_render_template_inline_template(self):
...
+        target_records = (self.render_object + self.render_object_fr).with_user(self.env.user)
+        with patch('odoo.models.Model.get_base_url', new=_mock_get_base_url):
+            render_results = self.env['mail.render.mixin']._render_template(
+                '<a href="/test/destination"><img src="/test/image"></a>',
+                target_records._name,
+                target_records.ids,
+                engine='inline_template',
+                post_process=True,
+            )
+        for partner_id, render_result in render_results.items():
```
Comment (tde-banana-odoo) [inline addons/mail/tests/test_mail_render.py:228]: Could be interesting to have some query counter checks, and be sure the prefetch effectively works

File: addons/product_email_template/tests/test_account_move.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ def setUp(self):
         self.template = Template.create({
             'name': 'Product Template',
             'subject': 'YOUR PRODUCT',
-            'model_id': self.env['ir.model']._get_id('product.template')
+            'model_id': self.env['ir.model']._get_id('account.move')
```
Comment (tde-banana-odoo) [inline addons/product_email_template/tests/test_account_move.py:13]: :sweat_smile:

**Review by tde-banana-odoo — COMMENTED** [review]
Zboing !

File: addons/mail/tests/test_mail_render.py (hunk 1 above)
Comment (reth-odoo) [reply addons/mail/tests/test_mail_render.py:233]: Indeed, might get expensive if it doesn't ^^

Added a query count check above, and a test commit to add the test before fixing. For documentation purposes.

Comment (tde-banana-odoo) [reply addons/mail/tests/test_mail_render.py:233]: No query counters before / after, to see the difference ?

Comment (reth-odoo) [reply addons/mail/tests/test_mail_render.py:233]: True

1 (fetch parameter) -> 6 (check access rights, rules and companies + fetch name)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**